### ISSUE TITLE
The choice of the values of the batch size

### ISSUE THREAD

**Issue by abdou31** [tensorflow/tensorflow#33574]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Windows 10 x64
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device: 
- TensorFlow installed from (source or binary): binary
- TensorFlow version (use command below):1.13.1
- Python version:3.6.4
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:9.0/7.5
- GPU model and memory:Nvidia Geforce 840m

**Describe the current behavior**
Scientists over the past years have made significant progress in increasing the speed of neural networks by dividing the data into different patches and their number in the thousands or sometimes millions of images and other data
The splitting into batches makes the neural network more efficient, but .....
In spite of this, the large size of the patches leads to weak results of the neural network often
It is not clear why the size of patches is sometimes appropriate when it is large and sometimes it is appropriate when it is small
But after research, it was concluded that the gradient noise scale can be useful in determining the appropriate patch size
Since it calculates the noise size approximate to the size of the patch and so it can calculate the size of variation in our data from the model point of view

When the noise scale is small, checking and studying a lot of data in parallel becomes fast, while when its volume is large, we can still learn a lot from large batches.
I have a dataset and I would like to train my own dataset to detect the eye region of human with landmarks.
In order to perform a good prediction after training , I would like to know if is it possible to calculate the gradient noise scale and adjust this variable with Tensorflow library to get a good value of the batch size?

**Code to reproduce the issue**
I used this simple CNN to detect the eye region:
https://github.com/yinguobing/cnn-facial-landmark

Thanks.

**Comment by rmothukuru** [comment]
This question is better asked on [StackOverflow](http://stackoverflow.com/questions/tagged//tensorflow) since it is not a bug or feature request. There is also a larger community that reads questions there.

**Comment by abdou31** [comment]
I have asked this question because I have a bug performance ( the avg loss of the training decrease and increase )

**Comment by gowthamkpr** [comment]
@abdou31 Can you please provide a small reproducible test case for us to reproduce your bug. Thanks!

**Comment by abdou31** [comment]
I have used a Github project "CNN facial landmark" for detecting eye region ( Resnet Architecture ).
I have trained my model with a dataset ( annotated with 40 points ).
That was the graph loss:

![image](https://user-images.githubusercontent.com/19480228/67637519-aae26200-f8db-11e9-9361-e0ed2c8a9e17.png)

I have tested the model on TestSet images and the result given is below:

The blue points: Predictions points
The green points: Annotations points

![image](https://user-images.githubusercontent.com/19480228/67637533-c51c4000-f8db-11e9-9077-1e2345c6a23d.png)

That was the code for testing the image:

```python
def _predict_input_fn():
    """Function for predicting."""
    return input_fn(
        record_file="./data300VW_test.record",
        batch_size=2,
        num_epochs=1,
        shuffle=False)


def serving_input_receiver_fn():
    """An input receiver that expects a serialized tf.Example."""
    image = tf.placeholder(dtype=tf.uint8,
                           shape=[IMG_HEIGHT, IMG_WIDTH, IMG_CHANNEL],
                           name='input_image_tensor')
    receiver_tensor = {'image': image}
    feature = tf.reshape(image, [1, IMG_HEIGHT, IMG_WIDTH, IMG_CHANNEL])
    return tf.estimator.export.ServingInputReceiver(feature, receiver_tensor)


def main(unused_argv):
    """MAIN"""
    # Create the Estimator
    estimator = tf.estimator.Estimator(
        model_fn=cnn_model_fn, model_dir="./tr2")

    # Choose mode between Train, Evaluate and Predict
    mode_dict = {
        'train': tf.estimator.ModeKeys.TRAIN,
        'eval': tf.estimator.ModeKeys.EVAL,
        'predict': tf.estimator.ModeKeys.PREDICT
    }

    mode = mode_dict['predict']

```

I have freezed the model using this code:

```python

from tensorflow.python.tools import freeze_graph

input_meta_graph = "C:\\Users\\test\\cnn-facial-landmark\\tr2\\model.ckpt-187690.meta"
checkpoint = 'C:\\Users\\test\\cnn-facial-landmark\\tr2\\model.ckpt-187690'
output_graph_filename ='C:\\Users\\test\\cnn-facial-landmark\\frozen_model_300VW.pb'

input_saver_def_path = ""
input_binary = True
output_node_names = 'logits/BiasAdd,input_to_float'
restore_op_name = "save/restore_all"
filename_tensor_name = "save/Const:0"
clear_devices = False

freeze_graph.freeze_graph(
"", input_saver_def_path, input_binary, checkpoint,
output_node_names, restore_op_name, filename_tensor_name,
output_graph_filename, clear_devices, "", "", "", input_meta_graph)

```

and I test the frozen model on a video, the result was so bad as you can see here;

![image](https://user-images.githubusercontent.com/19480228/67637580-15939d80-f8dc-11e9-8a1d-95e788b36c1d.png)


Can you help me to find the source of this problem?

**Comment by gowthamkpr** [comment]
@abdou31 Please post this question in stack overflow as it not related to bug/performance, build/installation or feature request. Thanks!

**Comment by abdou31** [comment]
I have solved the problem 
Thanks.
